tests/basicScenarioTest.m

```diff
@@ -0,0 +1,153 @@
+			testCase.Config.Harq.processes = 8;
+            testCase.Config.Harq.timeout = 3;
+            testCase.Config.Arq.active = true;
+			testCase.Config.Arq.maxRetransmissions = 1;
+			testCase.Config.Arq.maxBufferSize = 1024;
+            testCase.Config.Arq.timeout = 20;
+            
+            testCase.Simulation = Monster(testCase.Config);
+        end
+    end
+
+    methods (TestMethodTeardown)
+        function resetObjects(testCase)
+            %Test Teardown
+            clear all;
+        end
+    end
+
+    methods (Test)
+
+
+        function testResults(testCase)
+            %Run simulation
+            for iRound = 0:(testCase.Config.Runtime.totalRounds - 1)
+                testCase.Simulation.setupRound(iRound);
+                testCase.Simulation.run();          
+                testCase.Simulation.collectResults();
+                testCase.Simulation.clean();
+            end
+            %Verify results
+            %Verify utilization
+            testCase.verifyTrue( mean(testCase.Simulation.Results.util(2:end)) ==100); %First round util is 0, so that is omitted
+            %verify power consumption and state
+            %Power is calculated as CellReffP*P0+DeltaP*Pmax. This is 224W.
+            arrayfun(@(x) testCase.verifyTrue(x == 224), testCase.Simulation.Results.powerConsumed);
+            arrayfun(@(x) testCase.verifyEqual( x , 1), testCase.Simulation.Results.powerState); % powerState should be 1 in this case as it is always on
+            %verify scheduling
+            for i=2:testCase.Config.Runtime.numRounds
+                for j=1:testCase.Config.MacroEnb.numPRBs 
+                    testCase.verifyEqual(testCase.Simulation.Results.schedule(i,1,j).UeId, 1); %Assert that UE 1 is scheduled all the time
+                    testCase.verifyEqual(testCase.Simulation.Results.schedule(i,1,j).NDI, true); %Asssert that New Data Indicator indicates that there are more data.
+                    %testCase.verifyTrue(testCase.Simulation.Results.schedule(i,1,j).ModOrd == 2 ||...
+                    %testCase.Simulation.Results.schedule(i,1,j).ModOrd == 4 ||...
+                    %testCase.Simulation.Results.schedule(i,1,j).ModOrd == 6 ); %TODO: examine how to assert this
+                    %testCase.verifyEqual(testCase.Simulation.Results.schedule(i,1,j).Mcs > 20); %TODO: examine how to assert this
+                end
+            end
+            %verify HARQ and ARQ
+            arrayfun(@(x) testCase.verifyEqual(x ,0), testCase.Simulation.Results.harqRtx); %Should be 0. With good signal strength retransmission should be 0
+            arrayfun(@(x) testCase.verifyEqual(x ,0), testCase.Simulation.Results.arqRtx); %TODO: verify this statement
+            %verify BER and BLER
+            testCase.verifyTrue(mean(testCase.Simulation.Results.ber(2:end)) < 0.15 ); %Should be less than a certain threshhold. With good signal strength retransmission should be 0
```

Comment:
Maybe we could integrate this with the theoretical BER/BLER values computed based on SNR?

Reply:
Any good ideas of how to calculate this in theory?